### ISSUE TITLE
Pthread-backed ThreadParker in WordLock does work than needed.

### ISSUE THREAD

**Issue by thomcc** [Amanieu/parking_lot#272]
Specifically, unless I'm terribly confused, it never uses any of the timed functions, so AFAICT there's no need to make the condvar use `CLOCK_MONOTONIC`.

This saves needing to go through this hassle: https://github.com/Amanieu/parking_lot/blob/dac08482fec1297c15e67c2fe034a1a5c463915d/core/src/thread_parker/unix.rs#L137-L147

I don't know how much this changes the calculus of `IS_CHEAP_TO_CONSTRUCT`.

This is a little bit of a pain to fix since that stuff is still definitely needed by the non-wordlock code, though.

**Comment by bjorn3** [comment]
`pthread_cond_timedwait` is used:

https://github.com/Amanieu/parking_lot/blob/dac08482fec1297c15e67c2fe034a1a5c463915d/core/src/thread_parker/unix.rs#L97

**Comment by thomcc** [comment]
Right, but mean the code using the parkers here (e.g. WordLock) doesn't use park_until https://github.com/Amanieu/parking_lot/blob/dac08482fec1297c15e67c2fe034a1a5c463915d/core/src/word_lock.rs#L17
